src/libsyntax_pos/symbol.rs

```diff
@@ -530,37 +536,37 @@ where
     str: std::convert::AsRef<U>
 {
     fn as_ref(&self) -> &U {
-        self.string.as_ref()
+        self.get().as_ref()
     }
 }
 
 impl<T: std::ops::Deref<Target = str>> std::cmp::PartialEq<T> for LocalInternedString {
     fn eq(&self, other: &T) -> bool {
-        self.string == other.deref()
+        self.get() == other.deref()
     }
 }
 
 impl std::cmp::PartialEq<LocalInternedString> for str {
     fn eq(&self, other: &LocalInternedString) -> bool {
-        self == other.string
+        self == other.get()
     }
 }
 
 impl<'a> std::cmp::PartialEq<LocalInternedString> for &'a str {
     fn eq(&self, other: &LocalInternedString) -> bool {
-        *self == other.string
+        *self == other.get()
     }
 }
 
 impl std::cmp::PartialEq<LocalInternedString> for String {
     fn eq(&self, other: &LocalInternedString) -> bool {
-        self == other.string
+        self == other.get()
     }
 }
 
 impl<'a> std::cmp::PartialEq<LocalInternedString> for &'a String {
```

Comment:
There are too many of these, maybe a macro should be used (which can do `*other == *self`, I think).